File: schemas/manager_daily_brief.py
"""Manager daily brief — metadata-only decision support for Care Hub and briefing page."""

from __future__ import annotations

from typing import Any, Literal

from pydantic import BaseModel, ConfigDict, Field

ManagerDailyBriefScopeType = Literal["home", "user", "provider"]
ManagerDailyBriefTone = Literal["neutral", "attention", "urgent", "positive"]


class ManagerDailyBriefScope(BaseModel):
    model_config = ConfigDict(extra="ignore")

    type: ManagerDailyBriefScopeType = "provider"
    home_id: int | None = None
    user_id: str | None = None
    provider_id: int | None = None


class ManagerDailyBriefItem(BaseModel):
    model_config = ConfigDict(extra="ignore")

    id: str
    title: str
    safe_summary: str
    priority: Literal["low", "medium", "high", "urgent"] = "medium"
    route: str
    action_label: str | None = None
    source: str = "recording_alerts"
    child_id: int | None = None
    child_name: str | None = None
    metadata: dict[str, Any] = Field(default_factory=dict)


class ManagerDailyBriefSection(BaseModel):
    model_config = ConfigDict(extra="ignore")

    id: str
    title: str
    summary: str
    items: list[ManagerDailyBriefItem] = Field(default_factory=list)
    route: str
    action_label: str | None = None
    tone: ManagerDailyBriefTone = "neutral"
    metadata: dict[str, Any] = Field(default_factory=dict)


class ManagerDailyBriefRoutes(BaseModel):
    model_config = ConfigDict(extra="ignore")

    alerts: str = "/record/alerts"
    reviews: str = "/record/reviews"
    governance: str = "/record/governance"
    actions: str = "/actions"
    handover: str = "/handover/current"
    briefing: str = "/command-centre/briefing"
    care_hub: str = "/command-centre"
    orb: str = "/assistant/orb?mode=manager_daily_brief"
    isn: str = "/safeguarding"
    isn_orb: str = "/assistant/orb?mode=safeguarding_themes"


class ManagerDailyBrief(BaseModel):
    model_config = ConfigDict(extra="ignore")

    generated_at: str
    date: str
    scope: ManagerDailyBriefScope
    title: str = "Manager daily brief"
    opening_summary: str = ""
    recording_summary: str = ""
    review_summary: str = ""
    safeguarding_summary: str = ""
    isn_summary: str = ""
    action_summary: str = ""
    child_journey_summary: str = ""
    handover_summary: str = ""
    sections: list[ManagerDailyBriefSection] = Field(default_factory=list)
    recommendations: list[str] = Field(default_factory=list)
    limitations: list[str] = Field(default_factory=list)
    privacy_notice: str = (
        "This brief uses metadata and flags only — not full record bodies. "
        "It supports oversight; it does not make safeguarding threshold decisions "
        "or claim inspection compliance."
    )
    orb_prompts: list[dict[str, str]] = Field(default_factory=list)
    routes: ManagerDailyBriefRoutes = Field(default_factory=ManagerDailyBriefRoutes)
    metadata: dict[str, Any] = Field(default_factory=dict)
    reviewed: bool = False
    reviewed_at: str | None = None


class ManagerDailyBriefHealth(BaseModel):
    model_config = ConfigDict(extra="ignore")

    status: str = "ok"
    service: str = "manager_daily_brief_service"
    storage_mode: str = "memory"
    persistence_available: bool = False


class ManagerDailyBriefReviewRequest(BaseModel):
    model_config = ConfigDict(extra="ignore")

    date: str | None = None
    note: str | None = None


class ManagerDailyBriefReviewResponse(BaseModel):
    model_config = ConfigDict(extra="ignore")

    ok: bool = True
    reviewed: bool = True
    reviewed_at: str
    date: str
    message: str = "Daily brief marked as reviewed for today."

File: frontend-next/lib/os-api/manager-daily-brief.ts
export type ManagerDailyBriefItem = {
  id: string
  title: string
  safe_summary: string
  priority: 'low' | 'medium' | 'high' | 'urgent'
  route: string
  action_label?: string | null
  source: string
  child_id?: number | null
  child_name?: string | null
  metadata?: Record<string, unknown>
}

export type ManagerDailyBriefSection = {
  id: string
  title: string
  summary: string
  items: ManagerDailyBriefItem[]
  route: string
  action_label?: string | null
  tone: 'neutral' | 'attention' | 'urgent' | 'positive'
  metadata?: Record<string, unknown>
}

export type ManagerDailyBrief = {
  generated_at: string
  date: string
  scope: { type: string; home_id?: number | null; user_id?: string | null; provider_id?: number | null }
  title: string
  opening_summary: string
  recording_summary: string
  review_summary: string
  safeguarding_summary: string
  isn_summary?: string
  action_summary: string
  child_journey_summary: string
  handover_summary: string
  workforce_summary?: string
  sccif_summary?: string
  sections: ManagerDailyBriefSection[]
  recommendations: string[]
  limitations: string[]
  privacy_notice: string
  orb_prompts: Array<{ label: string; mode: string; query: string }>
  routes: {
    alerts: string
    reviews: string
    governance: string
    actions: string
    handover: string
    handover_reviews: string
    briefing: string
    care_hub: string
    orb: string
    isn?: string
    isn_orb?: string
  }
  metadata?: Record<string, unknown>
  reviewed?: boolean
  reviewed_at?: string | null
}

type ApiEnvelope<T> = { success?: boolean; data?: T }

const EMPTY_BRIEF: ManagerDailyBrief = {
  generated_at: '',
  date: '',
  scope: { type: 'user' },
  title: 'Manager daily brief',
  opening_summary: '',
  recording_summary: '',
  review_summary: '',
  safeguarding_summary: '',
  action_summary: '',
  child_journey_summary: '',
  handover_summary: '',
  sections: [],
  recommendations: [],
  limitations: [],
  privacy_notice: '',
  orb_prompts: [],
  routes: {
    alerts: '/record/alerts',
    reviews: '/record/reviews',
    governance: '/record/governance',
    actions: '/actions',
    handover: '/handover',
    handover_reviews: '/handover/reviews',
    briefing: '/command-centre/briefing',
    care_hub: '/command-centre',
    orb: '/assistant/orb?mode=manager_daily_brief'
  }
}

async function parseEnvelope<T>(response: Response, fallback: T): Promise<{ data: T; ok: boolean }> {
  if (!response.ok) return { data: fallback, ok: false }
  const payload = (await response.json().catch(() => ({}))) as ApiEnvelope<T> | T
  const envelope = payload as ApiEnvelope<T>
  return {
    data: envelope && typeof envelope === 'object' && 'data' in envelope ? (envelope.data as T) : (payload as T),
    ok: true
  }
}

export async function getManagerDailyBriefHealth() {
  const response = await fetch('/api/manager-daily-brief/health', { credentials: 'include', cache: 'no-store' })
  return parseEnvelope(response, { status: 'unavailable', service: 'manager_daily_brief_service' })
}

export async function getManagerDailyBrief() {
  const response = await fetch('/api/manager-daily-brief', { credentials: 'include', cache: 'no-store' })
  return parseEnvelope(response, EMPTY_BRIEF)
}

export async function markManagerDailyBriefReviewed(body?: { date?: string; note?: string }) {
  const response = await fetch('/api/manager-daily-brief/mark-reviewed', {
    method: 'POST',
    credentials: 'include',
    headers: { 'Content-Type': 'application/json' },
    body: JSON.stringify(body || {})
  })
  return parseEnvelope(response, { ok: false, reviewed: false, reviewed_at: '', date: '', message: '' })
}

/** Operational ORB only — never pass brief payload in URL. */
export function managerBriefOrbHref(mode: string, query?: string) {
  const params = new URLSearchParams({ mode })
  if (query) params.set('q', query)
  return `/assistant/orb?${params.toString()}`
}
